database/src/db.py

```diff
@@ -1,35 +1,123 @@
 import os
 import json
 import openai
-from pinecone import Pinecone
+from pinecone import Pinecone, ServerlessSpec
 from dotenv import load_dotenv
 
 
 class DatabaseHelper:
-    def __init__(self):
+    def __init__(self, index_name=None):
         # load .env file
         current_dir = os.path.dirname(os.path.abspath(__file__))
         env_path = os.path.join(current_dir, "..", "..", ".env")
         load_dotenv(env_path)
 
         # Configure APIs
-        pc = Pinecone(api_key=os.getenv("PINECONE_API_KEY"))
+        self.pc = Pinecone(api_key=os.getenv("PINECONE_API_KEY"))
         openai.api_key = os.getenv("OPENAI_API_KEY")
 
-        # Setup index
-        index_name = os.getenv("PINECONE_INDEX_NAME")
-        if not pc.has_index(index_name):
-            print("Index not found")
-        self.index = pc.Index(os.getenv("PINECONE_INDEX_NAME"))
-        self.describe_index()
+        # Setup index - use provided index name or get from env
+        self.index_name = index_name or os.getenv("PINECONE_INDEX_NAME")
+        
+        if not self.pc.has_index(self.index_name):
+            print(f"Index '{self.index_name}' not found")
```

Comment:
```suggestion
            print(f"Index '{self.index_name}' not found")
            self.index = None
            
```
can we initialize `self.index` here with `None` when there's no index found?